### DISCUSSION TITLE
Handle state changes in wizard

### DISCUSSION THREAD

**Discussion by hiimnhan** [cloudscape-design/components#687] [Q&A] [answered]
Let's say I have a table in step 2 of the Wizard. In that table I fetch data from BE and use useCollection hooks to load data to the table. I also allow user to inline edit and I use useReducer to store changes. However, when I jump to next step and back to that one, it will reuse data from server (since React mount the component again) and changes are not applied.

My question is what is a good way to handle things like that. I mean handle state changes in wizard components like selected items, inline editing...

Thank you!

**Comment by YueyingLu** [comment, marked as answer]
Hi @hiimnhan ,
thanks for reaching out! 
You might be interested to check our [demo page for wizard ](ttps://cloudscape.design/examples/react/wizard.html), also [the source code](https://github.com/cloudscape-design/demos/tree/main/src/pages/wizard). Hope it could help. 

Regards,
Yueying
